packages/test/test-end-to-end-tests/src/test/gcDataStoreRequests.spec.ts

```diff
@@ -66,51 +68,85 @@ describeNoCompat("GC Data Store Requests", (getTestObjectProvider) => {
+    let summaryCollection: SummaryCollection;
+
+    /**
+     * Waits for a summary that contains the op with given sequence number.
+     * @returns the version of this summary. This version can be used to load a Container with the summary associated
+     * with it.
+     */
+     async function waitForSummary(sequenceNumber: number): Promise<string> {
+        const ackedSummary: IAckedSummary = await summaryCollection.waitSummaryAck(sequenceNumber);
+        return ackedSummary.summaryAckNack.contents.handle;
+    }
 
-    const createContainer = async (): Promise<IContainer> => provider.createContainer(runtimeFactory);
-    const loadContainer = async (): Promise<IContainer> => provider.loadContainer(runtimeFactory);
-
-    async function waitForSummary(container: IContainer): Promise<string | undefined> {
-        let handle: string | undefined;
-        // wait for summary ack/nack
-        await new Promise((resolve, reject) => container.on("op", (op) => {
-            if (op.type === "summaryAck") {
-                handle = op.contents.handle;
-                resolve(true);
-            } else if (op.type === "summaryNack") {
-                reject(new Error("summaryNack"));
-            }
-        }));
-        return handle;
+    /**
+     * Wait for an op with the given key in the given shared directory.
+     * @returns the sequence number of this op.
+     */
+    async function waitForDirectoryOp(directory: ISharedDirectory, key: string): Promise<number> {
+        let sequenceNumber: number | undefined;
+        await new Promise((resolve) => {
+            const listener = (op: ISequencedDocumentMessage) => {
+                if (op.contents.key === key) {
+                    sequenceNumber = op.sequenceNumber;
+                    resolve(true);
+                    directory.off("op", listener);
+                }
+            };
+            directory.on("op", listener);
+        });
+        assert(sequenceNumber !== undefined);
+        return sequenceNumber;
     }
 
+    const createContainer = async (): Promise<IContainer> => provider.createContainer(runtimeFactory);
+    const loadContainer = async (summaryVersion: string): Promise<IContainer> => {
+        const requestHeader = {
+            [LoaderHeader.version]: summaryVersion,
+        };
+        return provider.loadContainer(runtimeFactory, undefined /* options */, requestHeader);
+    };
+
     beforeEach(async () => {
         provider = getTestObjectProvider();
 
         // Create a Container for the first client.
-        container1 = await createContainer();
+        mainContainer = await createContainer();
+
+        // Create and setup a summary collection that will be used to track and wait for summaries.
+        summaryCollection = new SummaryCollection(
+            mainContainer.deltaManager.initialSequenceNumber,
+            new TelemetryNullLogger(),
+        );
+        mainContainer.deltaManager.inbound.on("op",
```

Comment:
@anthony-murphy I will update this once your PR #5739 is merged.

Reply:
i merged it